dataset/batch_image.py

```diff
@@ -1,525 +1,717 @@
+        ----------
+        dst : str, None
+            Path to the folder where to dump. If dst is None then path is determined from index.
+        fmt : {'image', 'blosc', 'csv', 'hdf5', 'feather'}
+            Format of the file to save.
+        components : str, sequence
+            components to save. Note that if `fmt='images'` than components must be str.
+
+        Returns
+        -------
+        self
+        """
+        if fmt == 'image':
+            self._dump_image(dst, components)
+        return super().dump(dst, fmt, components, *args, **kwargs)
 
-        p : float
-            a probability to apply rotate
-            (0. - don't rotate, .5 - rotate half of images, 1 - rotate all images)
 
-        angle : tuple
-            an angle range in the form of (min_angle, max_angle), in degrees
-        """
-        if np.random.binomial(1, p) > 0:
-            angle = angle or (-45., 45.)
-            _angle = np.random.uniform(*angle)
-            preserve_shape = kwargs.pop('preserve_shape', True)
-            return self._rotate_one(ix, components, _angle, preserve_shape=preserve_shape, **kwargs)
-        image = self.get(ix, components)
-        return image
+@transform_actions(prefix='_', suffix='_all', wrapper='apply_transform_all')
+@transform_actions(prefix='_', suffix='_', wrapper='apply_transform')
+class ImagesBatch(BaseImagesBatch):
+    """ Batch class for 2D images.
 
-    @action
-    def flip(self, axis=None):
-        """ Flip images
+    Images are stored as numpy arrays (N, H, W, C).
+    """
+
+    @property
+    def image_shape(self):
+        """: tuple - shape of the image """
+        return self.images.shape[1:]
+
+    @inbatch_parallel(init='indices', post='assemble', target='async')
+    def _load_image(self, ix, src=None, components="images"):
+        """ Wrapper for scipy.ndimage.open.
+
+        .. note:: only works with a single component
 
         Parameters
         ----------
-        axis : {'h', 'v'}
-            'h' for horizontal (left/right) flip
-            'v' for vertical (up/down) flip
-        direction : {'l', 'r', 'u', 'd'}
+        path : str, None
+        ix : str
+            element's index (filename)
+        components : str
+            component to load
         """
-        if axis == 'h':
-            return self.fliplr()
-        elif axis == 'v':
-            return self.flipud()
-        else:
-            raise ValueError("Parameter axis can be 'h' or 'v' only")
+        return scipy.ndimage.open(self._make_path(src, ix))
 
+    @inbatch_parallel(init='indices', target='async')
+    def _dump_image(self, ix, dst=None, components='images'):
+        """ Save image to dst.
 
-class ImagesBatch(BaseImagesBatch):
-    """ Batch class for 2D images
+        Actually a wrapper for scipy.misc.imsave.
 
-    images are stored as numpy arrays (N, H, W) or (N, H, W, C)
+        .. note:: `components` must be str
```

Comment:
To many formats and texts for the very same message.

Make it a note everywhere with the same text.

Reply:
+